backend/src/main/java/com/eshka/service/SubjectService.java

```diff
@@ -8,9 +8,24 @@
 @Service
 @RequiredArgsConstructor
 public class SubjectService {
-    private SubjectRepository subjectRepository;
+    private final SubjectRepository subjectRepository;
 
     public Subject findById(long id) {
         return subjectRepository.findById(id).orElseThrow();
     }
+
+    public Subject createNewSubject(Subject subject) {
+        return subjectRepository.save(subject);
+    }
+
+    public Subject editSubject(Subject subject) {
+        Subject oldSubject = subjectRepository.findById(subject.getId()).orElseThrow();
```

Comment:
What exception will be thrown? Do you handle it somewhere? If not, please, add custom exceprion with handling.